bgpd/bgp_bfd.c

```diff
@@ -635,6 +663,73 @@ DEFUN_HIDDEN (neighbor_bfd_type,
 	return CMD_SUCCESS;
 }
 
+static int bgp_bfd_set_check_controlplane_failure_peer(struct vty *vty, struct peer *peer,
+						       const char *no)
+{
+	struct bfd_info *bfd_info;
+
+	if (!peer->bfd_info) {
+		if (no)
+			return CMD_SUCCESS;
+		vty_out(vty, "%% Specify bfd command first\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+	bfd_info = (struct bfd_info *)peer->bfd_info;
+	if (!no) {
+		if (!CHECK_FLAG(bfd_info->flags, BFD_FLAG_BFD_CHECK_CONTROLPLANE)) {
+			SET_FLAG(bfd_info->flags,  BFD_FLAG_BFD_CHECK_CONTROLPLANE);
+			bgp_bfd_update_peer(peer);
+		}
+	} else {
+		if (CHECK_FLAG(bfd_info->flags, BFD_FLAG_BFD_CHECK_CONTROLPLANE)) {
+			UNSET_FLAG(bfd_info->flags,  BFD_FLAG_BFD_CHECK_CONTROLPLANE);
+			bgp_bfd_update_peer(peer);
+		}
+	}
+	return CMD_SUCCESS;
+}
+
+
+DEFUN (neighbor_bfd_check_controlplane_failure,
+       neighbor_bfd_check_controlplane_failure_cmd,
+       "[no] neighbor <A.B.C.D|X:X::X:X|WORD> bfd check-control-plane-failure",
```

Comment:
I assume this could be extended to support IS-IS or OSPF in the future (?)... I think this command will still be okay, just wanted to bring this up so it's "thought about." :-)

Reply:
I did not find some pointers related to usage of ospf graceful restart and bfd, but if you have, do not hesitate.